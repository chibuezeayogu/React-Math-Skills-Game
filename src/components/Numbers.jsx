import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';


/**
 * 
 * @class Numbers
 * @extends Component
 */
class Numbers extends Component {
  /**
   * 
   * @method selectNumber
   * 
   * @memberof Game
   * 
   * @param {integer} number
   * 
   * @returns {void}
   */
  numberClassName = (number) => {
    if (this.props.usedNumbers.indexOf(number) >= 0) return 'used';
    
    if (this.props.selectedNumbers.indexOf(number) >= 0) return 'selected';
  };

  /**
   * 
   * @method handleSelectNumber
   * 
   * @memberof Game
   * 
   * @param {object} event
   * 
   * @returns {void}
   */
  handleSelectNumber = (event) => {
    this.props.selectNumber(event.target.id);
  }

  render() {
    return (
      <div>
          {
            Numbers.list.map((number, i) => <span 
              key={i} 
              className={this.numberClassName(number)}
              id={number}
              role="button"
              tabIndex={0} 
              onClick={this.handleSelectNumber}
            >
              {number}
            </span>
            )
          }
      </div>
    );
  }
}

Numbers.list = _.range(1, 10);

Numbers.propTypes = {
  selectedNumbers: PropTypes.array,
  selectNumber: PropTypes.func,
  usedNumbers: PropTypes.array
};

export default Numbers;
