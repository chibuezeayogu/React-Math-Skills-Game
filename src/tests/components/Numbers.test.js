/* 
global jest 
expect 
*/
import React from 'react';
import { mount } from 'enzyme';
import Numbers from '../../components/Numbers';

describe('Numbers Component', () => {
  const setup = () => {
    const props = {
      selectedNumbers: [1, 2, 3, 4, 5, 6, 7, 8],
      selectNumber: jest.fn(),
      usedNumbers: [9]
    };

    return mount(<Numbers {...props}/>);
  };

  const wrapper = setup();

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });


  it('should selete a selectNumber', () => {
    wrapper.find('#selectNumber').first().simulate('click');
  });
});
