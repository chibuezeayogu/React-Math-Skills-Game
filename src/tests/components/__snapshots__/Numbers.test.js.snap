// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Numbers Component should render correctly 1`] = `
<div>
  <span
    className="selected"
    id={1}
    key="0"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    1
  </span>
  <span
    className="selected"
    id={2}
    key="1"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    2
  </span>
  <span
    className="selected"
    id={3}
    key="2"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    3
  </span>
  <span
    className="selected"
    id={4}
    key="3"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    4
  </span>
  <span
    className="selected"
    id={5}
    key="4"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    5
  </span>
  <span
    className="selected"
    id={6}
    key="5"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    6
  </span>
  <span
    className="selected"
    id={7}
    key="6"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    7
  </span>
  <span
    className="selected"
    id={8}
    key="7"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    8
  </span>
  <span
    className="used"
    id={9}
    key="8"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    9
  </span>
</div>
`;
